stcrpy/tcr_metrics/tcr_dockq.py

```diff
@@ -0,0 +1,405 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import warnings
+from collections import namedtuple
+import itertools
+import Bio
+import json
+from functools import partial
+import logging
+import numpy as np
+import copy
+import os
+
+
+class TCRDockQ:
+    def __init__(self, TCR_pMHC_interface=True, **kwargs):
+        """
+        Initialize the TCRDockQ class. See `Dockq --help` for more information.
+
+        Args:
+            TCR_pMHC_interface (bool): Whether to use evaluate the entire TCR-pMHC interface or evaluate spearate chains. Deaults to True. This will create a merged psudo-chain for the antigen and the TCR.
+            **kwargs: Keyword arguments from DockQ: mapping, small_molecule, capri_peptide, no_align, json, n_cpu, max_chunk, allowed_mismatches, verbose, short, print_results.
+        """
+
+        DockQArgs = namedtuple(
+            "DockQArgs",
+            [
+                "mapping",
+                "small_molecule",
+                "capri_peptide",
+                "no_align",
+                "json",
+                "n_cpu",
+                "max_chunk",
+                "allowed_mismatches",
+                "verbose",
+                "short",
+                "print_results",
+            ],
+        )
+        self.TCR_pMHC_interface = TCR_pMHC_interface
+        self.args = DockQArgs(
+            mapping=kwargs.get("mapping", None),
+            small_molecule=kwargs.get("small_molecule", None),
+            capri_peptide=kwargs.get("capri_peptide", False),
+            no_align=kwargs.get("no_align", False),
+            json=kwargs.get("json", None),
+            n_cpu=kwargs.get("n_cpu", 8),
+            max_chunk=kwargs.get("max_chunk", 512),
+            allowed_mismatches=kwargs.get("allowed_mismatches", 0),
+            verbose=kwargs.get("verbose", False),
+            short=kwargs.get("short", False),
+            print_results=kwargs.get("print_results", False),
+        )
+        return
+
+    def try_tcr_mapping(self, dock: "abTCR", reference: "abTCR"):
+        try:
+            dock_chain_mapping = {v: k for k, v in dock.get_chain_mapping().items()}
+            ref_chain_mapping = {v: k for k, v in reference.get_chain_mapping().items()}
+
+            mapping = [
+                (dock_chain_mapping[chain_type], ref_chain_mapping[chain_type])
+                for chain_type in dock_chain_mapping
+                if chain_type in ref_chain_mapping
+            ]
+            mapping = list(zip(*mapping))
+            string_mapping = f"{''.join(mapping[0])}:{''.join(mapping[1])}"
+            return mapping, string_mapping
+        except Exception:
```

Comment:
Could this be swapped to a more specific exception to catch? Exception allows a lot to be let through

Reply:
+1